theano/gof/params_type.py

```diff
@@ -191,7 +238,28 @@ def __init__(self, **kwargs):
         self.length = len(kwargs)
         self.fields = tuple(sorted(kwargs.keys()))
         self.types = tuple(kwargs[field] for field in self.fields)
-        self.name = self.generate_struct_name()
+        self.name = self.__generate_struct_name()
+
+        self.__const_to_enum = {}
+        self.__alias_to_enum = {}
+        enum_types = [t for t in self.types if isinstance(t, EnumType)]
+        if enum_types:
+            # We don't want same enum names in different enum types.
+            if sum(len(t) for t in enum_types) != len(set(k for t in enum_types for k in t)):
+                raise AttributeError('Wrapper: found different enum types with common constant names.')
+            # We don't want same aliases in different enum types.
+            if sum(len(t.aliases) for t in enum_types) != len(set(alias for t in enum_types for alias in t.aliases)):
+                raise AttributeError('Wrapper: found different enum types with common constant aliases.')
```

Comment:
I think we should also check that we don't have alias that have the same name as enums name